pkg/query-service/app/logs/v3/json_filter.go

```diff
@@ -106,6 +107,29 @@ func getJSONFilterKey(key v3.AttributeKey, op v3.FilterOperator, isArray bool) (
 	return keyname, nil
 }
 
+// takes the path and the values and generates where clauses for better usage of index
+func getPathIndexFilter(path string) string {
+	filters := []string{}
+	keyArr := strings.Split(path, ".")
+	if len(keyArr) < 2 {
+		return ""
+	}
```

Comment:
This `< 2` always becomes a puzzle. It seems like this is for JSON querying. Can we somehow make use of the `IsJSON` info instead of this split and length comparison?

Reply:
So we already have that check, this is just an extra independent check for this function.

https://github.com/SigNoz/signoz/blob/e639d79ba9864547217c45531d9f08569d461f7f/pkg/query-service/app/logs/v3/query_builder.go#L166

Reply:
The problem with this kind of code is that you understand things in your head but it's hard to reason about for everyone else. What do you think?

Reply:
yeah aggree, I think I should just remove this then as it is just creating confusion and the check is already done before

Reply:
I don't think I should remove this, instead I added a comment which explain this to future reviewers, hope it helps.